400-rest/src/test/java/software/wings/sm/states/azure/AzureVMSSStateHelperTest.java

```diff
@@ -1232,4 +1234,42 @@ public void testGetArtifactForRollback_onDemand() {
     assertThat(artifact.isPresent()).isTrue();
     assertThat(artifact.get()).isEqualTo(rollbackArtifact);
   }
+
+  @Test
+  @Owner(developers = OwnerRule.VAIBHAV_KUMAR)
+  @Category(UnitTests.class)
+  public void testMetadataOnlyArtifactStreamTypes() {
+    String serviceId = "serviceUUID";
+    String appId = "appId";
+    List<String> artifactPaths = Collections.singletonList("artifactPath");
+    ExecutionContext context = mock(ExecutionContext.class);
+    Artifact artifact = new Artifact();
+    ServiceElement serviceElement = ServiceElement.builder().uuid(serviceId).build();
+    PhaseElement phaseElement = PhaseElement.builder().serviceElement(serviceElement).build();
+
+    when(context.getAppId()).thenReturn(appId);
+    when(context.getContextElement(any(), any())).thenReturn(phaseElement);
+    when(serviceResourceService.getWithDetails(any(), any())).thenReturn(Service.builder().build());
+
+    // Case 1: JENKINS
+    // Expected behaviour: metadataOnly flag must be set to true
+    JenkinsArtifactStream jenkinsArtifactStream =
+        JenkinsArtifactStream.builder().artifactPaths(artifactPaths).metadataOnly(true).build();
+    when(artifactStreamService.get(any())).thenReturn(jenkinsArtifactStream);
+
+    ArtifactConnectorMapper artifactConnectorMapper = azureVMSSStateHelper.getConnectorMapper(context, artifact);
+
+    assertThat(artifactConnectorMapper.artifactStreamAttributes().isMetadataOnly()).isTrue();
+
+    // Case 2: DOCKER
+    // Expected behaviour: metadataOnly flag must be set to false
+    DockerArtifactStream dockerArtifactStream = DockerArtifactStream.builder().metadataOnly(true).build();
+    Service service = Service.builder().artifactType(ArtifactType.DOCKER).build();
+    when(artifactStreamService.get(any())).thenReturn(dockerArtifactStream);
+    when(serviceResourceService.getWithDetails(any(), any())).thenReturn(service);
+
+    artifactConnectorMapper = azureVMSSStateHelper.getConnectorMapper(context, artifact);
+
+    assertThat(artifactConnectorMapper.artifactStreamAttributes().isMetadataOnly()).isFalse();
```

Comment:
Let's use a different `artifactConnectorMapper` altogether, so that if the original `artifactConnectorMapper` gets modified/deleted, we wouldn't need to modify the reused `artifactConnectorMapper`.

Let's rename this too.

```suggestion
    ArtifactConnectorMapper  dockerArtifactConnectorMapper = azureVMSSStateHelper.getConnectorMapper(context, artifact);

    assertThat(dockerArtifactConnectorMapper.artifactStreamAttributes().isMetadataOnly()).isFalse();
```

Reply:
Renamed in 71f820e4c771a874324d4f15f6f07b427dd0defd.